extras/envoy-xds/pkg/discovery.go

```diff
@@ -0,0 +1,71 @@
+package pkg
+
+import (
+	"encoding/json"
+	"fmt"
+	_ "github.com/Jille/grpc-multi-resolver"
+	"github.com/flipkart-incubator/dkv/pkg/ctl"
+	"github.com/flipkart-incubator/dkv/pkg/serverpb"
+	"io/ioutil"
+	"log"
+)
+
+type DiscoveryClient struct {
+	client     *ctl.DKVClient
+	configPath string
+}
+
+// Create a connection to DKV Service discovery client which provides a view of entire cluster
+// Cluster comprises of all databases and shards (vBuckets) which are registering to this service discovery group
+func InitServiceDiscoveryClient(discoveryServiceAddr string, configPath string) *DiscoveryClient {
+	client, err := ctl.NewInSecureDKVClient(discoveryServiceAddr, "")
+	if (err != nil) {
+	   log.Panicf("Failed to start Discovery Client %v.", err)
+	}
+	return &DiscoveryClient{client: client, configPath: configPath}
+}
+
+func (c *DiscoveryClient) GetEnvoyConfig() (EnvoyDKVConfig, error) {
+
+	config, err := ioutil.ReadFile(c.configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	kvs := make(map[string]interface{})
+	err = json.Unmarshal(config, &kvs)
+	if err != nil {
+		return nil, err
+	}
+
+	regions, err := c.client.GetClusterInfo("", "", "")
```

Comment:
How would this work when a cluster has multiple shards/masters?

Reply:
updated dc id and database to be read from config. 

current client libraries are selecting only shard id and thus assuming dc id / database to be uniform. will need to change these contracts and support multi dc / multi database awareness from clients. scoping out of current PR as not a blocker for now

Reply:
Multi dc could be a future use-case, but each shard would be a different database otherwise DS would think that all of them are same resource hence unable to distinguish between them.

Reply:
why would each shard be a different database ? like session-service database has say 20 shards, athena has 10 shards. but they're different clusters and each cluster has only 1 database.

Reply:
Ah, my misunderstanding, `vbucket` would be used in this case.